packages/hash/frontend/src/components/hooks/use-loading-callback.ts

```diff
@@ -0,0 +1,40 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+
+const useIsMounted = () => {
+  const mounted = useRef<boolean>();
+
+  useEffect(() => {
+    mounted.current = true;
+
+    return () => {
+      mounted.current = false;
+    };
+  }, []);
+
+  return mounted;
+};
+
+export const useLoadingCallback = <
+  TCallback extends (...args: any[]) => Promise<unknown> | unknown,
+>(
+  callback: TCallback,
+): [TCallback, boolean] => {
+  const [loading, setLoading] = useState<boolean>(false);
+  const isMounted = useIsMounted();
+
+  const loadingCallback = useCallback(
+    async (...args: Parameters<TCallback>) => {
+      setLoading(true);
+      try {
+        return await callback(...args);
+      } finally {
+        if (isMounted.current) {
+          setLoading(false);
+        }
+      }
+    },
+    [callback, isMounted],
```

Comment:
I've just realised – this `useCallback` is useless because you're passing inline callbacks in and then using them in the dependency array.

Reply:
Removed this hook here https://github.com/hashintel/hash/pull/1734/commits/3fb7ea7e6cc8f2bfd17ece82ef8a9f5940f98fde